src/libtiled/mapobject.cpp

```diff
@@ -358,9 +358,12 @@ void MapObject::flip(FlipDirection direction, const QPointF &origin)
  */
 MapObject *MapObject::clone() const
 {
+    // TODO should I clone components
+    // how are custom properties cloned
```

Comment:
Properties, and also Components as you've introduced them, are referenced by value and cloning is just done by making a copy, as you've done here.

Of course, Qt containers (like `QMap`) use implicit sharing, which means a copy just increments a reference count, and a real deep copy only happens upon modification. But regarding the cloning that's just an implementation detail (though a useful one, since it makes the cloning cheaper when the clone is never modified).